File: js/examples/evals/example.eval.ts
import { Eval } from "braintrust";
import { Levenshtein } from "autoevals";

const NUM_EXAMPLES = 10000;

Eval("queue-test", {
  data: () => {
    const data: { input: string; expected: string }[] = [];
    for (let i = 0; i < NUM_EXAMPLES; i++) {
      const names = [
        "Foo",
        "Bar",
        "Alice",
        "Bob",
        "Charlie",
        "Diana",
        "Eve",
        "Frank",
      ];
      const greetings = ["Hi", "Hello", "Hey", "Greetings"];

      const name = names[i % names.length];
      const greeting = greetings[i % greetings.length];

      data.push({
        input: name,
        expected: `${greeting} ${name}`,
      });
    }
    return data;
  },
  task: async (input) => {
    return "Hi " + input; // Replace with your LLM call
  },
  scores: [Levenshtein],
});

File: py/examples/evals/example.eval.py
from braintrust import Eval

NUM_EXAMPLES = 10


def exact_match_scorer(input, output, expected):
    if expected is None:
        return 0.0
    return 1.0 if output == expected else 0.0


def data_fn():
    data = []
    for i in range(NUM_EXAMPLES):
        names = [
            "Foo",
            "Bar",
            "Alice",
            "Bob",
            "Charlie",
            "Diana",
            "Eve",
            "Frank",
        ]
        greetings = ["Hi", "Hello", "Hey", "Greetings"]

        name = names[i % len(names)]
        greeting = greetings[i % len(greetings)]

        data.append({"input": name, "expected": f"{greeting} {name}"})
    return data


def task_fn(input, hooks=None):
    return f"Hi {input}"


Eval(
    "queue-test",
    data=data_fn,
    task=task_fn,
    scores=[exact_match_scorer],
)
